% Version: final

\documentclass[../main.tex]{subfiles}

\begin{document}

\subsubsection{Models of gene regulatory networks}
% Refer to some nice sources for more information on sea urchin and GRN, bolouri is nice for latter, former [2] of kuhn

% The state of network simulation
%% What is a gene regulatory network?
The developmental growth of complex animals is driven by the spatial and temporal activation of gene transcription to mRNA, which drives the productions of gene products such as proteins.
This spatial and temporal sequence of states is determined in the genomic regulatory code of an animal \cite{bolouri2002modeling, kuhn2009monte}.
This code specifies gene regulatory networks (GRNs), which define activation- and suppression relationships between genes.
Understanding the GRNs that drive processes in cells is key to understanding how a fertilized egg, a single cell, can grow out to an animal, a large and complex symbiosis of billions of cells.
It can also help us understand the mechanisms behind some human diseases, such as cancer \cite{qian2008inference}.

%% How do we describe gene regulatory networks
Constructing models that capture a GRN accurately is a complicated process.
It is experimentally difficult to measure kinetic parameters associated with cellular processes in vivo \cite{bolouri2002modeling}.
It is important to decide on an appropriate level of complexity when choosing a model to represt a GRN system; models that are too complex lack in explanatory power, whereas models that are too simple cannot accurately represent the dynamics of the real system \cite{geard2009dynamical}.
Any model of a GRN system should have a way to describe both the state of the system, and the way the system changes over time \cite{geard2009dynamical}.

The most basic way to represent a network is by a full Boolean network \cite{bolouri2002modeling}.
This network maps relationships between genes as instantaneous 'switches', allowing the production of a crude model without in-depth knowledge of reaction rates and delays in the real system.
The state of a Boolean network is stored by tracking which genes are turned off and on \cite{geard2009dynamical}.
The changes of the system over time are stored in the form of a transition table.
Boolean networks are constructed through arrayed gene expressions assays, which cluster related genes, followed by a regulatory linkage analysis, which disrupts the activity of a gene to observe the impact on downstream genes in the network \cite{bolouri2002modeling, wu2013high}.

Typically, they are only accurate approximations for small networks \cite{karlebach2008modelling}.
Probabilistic elements can be incorporated in a Boolean network \cite{schlitt2007current}.
This is typically done when empirical data is lacking, and there is uncertainty about the relationships between genes in the network \cite{karlebach2008modelling}.
An extension upon probabilistic Boolean networks is the petrinet, which functions by mimicking the buildup of transcription products over several time steps \cite{karlebach2008modelling}.
Only when a 'bucket' fills, the down- or upregulation relationship is applied.
Boolean networks can also be extended by using multi-valued logic over Boolean logic \cite{geard2009dynamical}.
Boolean models work under strong assumptions, for instance that inhibition dominates over activation, which is often the case in real GRN systems \cite{he2016algorithm, wang2010process}. % remove second if published (will remove this anyway I guess)

Boolean motifs cannot accurately describe reality in all cases, as it has been shown that one Boolean motif can have vastly different functions based on different kinetic properties in a continuous model \cite{ingram2006network}.
Continuous rules are commonly captured in an ODE system or other algebraic formalism that describes the slow reactions involved in activators and suppressors binding to DNA, as well as transcription and translation \cite{ingram2006network}.
An ODE system is typically preferred, as this mimics the reaction dynamics in the cell more closely, but is also more difficult to construct than a simpler continuous model.
Fast reactions, such as protein-protein interactions, can still be modeled as Boolean switches in these models, as they occur at completely different timescales.
A continuous model can be constructed  from the Boolean counterpart, when additional research is done through the measurement of kinetic data, followed by verification to measure the correspondence of the network with reality \cite{bolouri2002modeling}.
It is preferable to use a non-linear ODE, as GRNs are non-linear in nature \cite{qian2008inference, tyson2003sniffers}.
A proposed improvement upon the common ODE models is that of a sparse additive ODE model, able to capture nonlinear relationships \cite{wu2014sparse}.

Other types of models exist, notably stochastic models, hidden Markov models and multi-valued Boolean models \cite{bolouri2002modeling, geard2009dynamical, wu2014sparse}. % , but these fall outside the scope of this research
Stochasticity can be introduced both in discrete and continuous models, and removes the assumption that product concentrations are not subject to outside influences, and that the control of these concentrations is deterministic \cite{geard2009dynamical}.
The latter is a problem as, especially when regulatory molecules are present in a low concentration, reaction rates can fluctuate without changes in the concentration of the regulatory compounds \cite{geard2009dynamical}.
In some cases, it is preferable to describe a GRN as a mixture of Boolean logic and continuous rules \cite{bolouri2002modeling}.

The parameters in continuous models can be notoriously difficult to fit to empirical data, as this requires the search through multi-dimensional space for an optimum fit \cite{bolouri2002modeling, kuhn2009monte}.
In addition, the number of parameters is large in ODE models, and the number of real systems for which we have accurate measurements of these parameters is small \cite{geard2009dynamical}.
This challenge has been tackled with some success using, amongst others, Monte Carlo methods and genetic programming paired with Kalman filtering \cite{qian2008inference, kuhn2009monte}.
More elaborate statistical methods have been developed as well, such as the modified elastic net-method, LASSO-methods and the Bayesian best subset regression \cite{greenfield2013robust, wu2014sparse}.

\subsubsection{Information theory and GRNs}

An inquiry into GRNs using information theory has been made as well, primarily at the level of common network motifs \cite{zhang2012chaotic}.
Positive feedback loops have been found to function as switches and memory units, whereas negative feedback loops have been found to have a noise suppressing or oscillation-inducing function.
Studies have been done into the prevalence of chaotic behavior in GRNs \cite{zhang2012chaotic}.
These resulted in the conclusion that chaotic motifs of size $n\ge 3$ exist, but that they are uncommon in real networks, as they require competition between multiple feedback loops, at least one of which should be a negative feedback loop \cite{zhang2012chaotic}.
In real networks, this condition is scarcely met, although some GRNs do meet this condition, most notably the $n=4$ GRN that regulates the P53-system.

\subsubsection{Available GRN models}

There is a modest selection of GRN models available for further research.
These models are typically released in the SBML-format, an enriched xml-datatype.
One of the best captured and most researched GRNs is the endomesoderm GRN of the sea urchin \cite{bolouri2002modeling, kuhn2009monte}.
This network describes the activity of gene regulation in the early development of the sea urchin embryo, and is still in the process of being updated as new studies are done \cite{urchinmodel}.
An attempt has been made to build a full ODE model based on the Boolean abstraction of this GRN using Monte Carlo methods.
This resulted in some success, as 65\% of the maximum possible correspondence with emperical data was achieved \cite{kuhn2009monte}.
In this reaction rate-type ODE model, the mRNA concentration linked to a gene (a measure for gene activity) is expressed as
%
\begin{equation}
\frac{dX}{dt} = (\frac{k_A \cdot A\left( t \right)}{c_A + A\left( t \right)} + \frac{k_B \cdot B\left( t \right)}{c_B + B\left( t \right)}) \cdot \frac{k_C \cdot C\left( t \right)}{c_C + C\left( t \right)} - k_\mathrm{deg} X\left( t \right)
\end{equation}
%
where $A$, $B$ and $C$ are protein concentrations, and the lower case letters denote kinetic constants.
Such a model of chemical master equations is popular, like sigmoidal and Michaelis-Menten models, as most limiting processes in gene regulation are chemical reactions \cite{aijo2009learning}.
The model contains of 54 genes, 140 variable species, 278 reactions and 287 parameters.

A model that puts less focus on mimicking real reaction rates to capture GRNs is a system of continuous non-linear differential equations.
This has been implemented by \cite{qian2008inference} as
%
\begin{equation}
\frac{dX_i}{dt} = f_i \left( X_1,...,X_n \right) + v_i
\end{equation}
%
where
%
\begin{equation}
f_i = \sum_{j=1}^{L_i} [(w_{ij}+ \mu_{ij})\Omega_{ij}\left( x_1,...,x_n\right)]
\end{equation}
%
and $\Omega\left( x_1,...,x_n\right)$ is a non-linear function, such as a sigmoid function.
However, no dataset was freely available from this study.

% Yeast model
% Skip the artificial NN for now, we are looking at motifs
Most models, both ODE and Boolean types, have been made to describe the yeast \textit{S. cerevisiae}, and more recently also \textit{S. pombe} \cite{ferrell2011modeling}.
As a result, using the \textit{S. cerevisiae} network is encouraged for most purposes, as this network is well validated, up-to-date with current technologies and widely accepted.

% Looking at a subsection
Due to the large size of GRNs, it is often not possible to apply computational methods on the network as a whole.
To circumvent this limitation, it is possible to examine functional motifs in the network, isolated from the rest.
This can be done simply by removing all nodes, saving only the nodes of interest \cite{zhang2012chaotic}.
When using probability distributions instead of set initial conditions, it is possible to capture the influence of the rest of the network on the input variables at the present moment by enforcing empirically determined correlations, thus drawing from a joint PDF.
However, the omitted part of the network is not considered performing an evolution over time.
This implies that isolated motifs should only be examined locally in time. % I could phrase this better
\end{document}