% Version: 0.0

\documentclass[../main.tex]{subfiles}

\begin{document}

\subsubsection{Describing gene regulatory networks}
%TODO
% Refer to some nice sources for more information on sea urchin and GRN, bolouri is nice for latter, former [2] of kuhn

% The state of network simulation
%% What is a gene regulatory network?
The developmental growth of complex animals is driven by the spatial and temporal activation of gene transcription to mRNA, and consequentially further gene products such as proteins.
This spatial and temporal sequence of states is determined in the genomic regulatory code of an animal \cite{bolouri2002modeling, kuhn2009monte}.
This code specifies gene regulatory networks (GRNs), which are networks of activation- and suppression relationships between genes.
Understanding the GRNs that drive processes in cells is key to understanding how a fertilized egg, a single cell, can grow out to an animal, a large and complex symbiosis of billions of cells.
It can also help us understand the mechanisms behind some human diseases, such as cancer.

%% How do we describe gene regulatory networks
Constructing models that capture a GRN accurately is a complicated process.
It is experimentally difficult to measure kinetic parameteres associated with cellular processes in vivo \cite{bolouri2002modeling}.
The most basic way to represent a network is by a full Boolean network \cite{bolouri2002modeling}.
This network maps relationships between genes as instantaneous 'switches', allowing the production of a crude model without in-depth knowledge of reaction rates and delays in the real system.
Boolean networks are constructed through arrayed gene expressions assays, which cluster related genes, followed by a regulatory linkage analysis, which disruptsthe activity of a gene to observe the impact on downstream genes in the network \cite{bolouri2002modeling}.
Typically, it is ideal to describe a GRN as a mixture of Boolean logic and continuous rules \cite{bolouri2002modeling}.
This is for the reason that Boolean motifs can have vastly different functions based on kinetic properties \cite{ingram2006network}.
Continuous rules are commonly captured in an ODE system or other algebraic formalism that describes the slow reactions involved in activators and surpressors binding to DNA, as well as transcription and translation \cite{\cite{ingram2006network}}.
An ODE system is typically preferred, as this mimics the reaction dynamics in the cell more closely, but is also more difficult to construct that a simpler continuous model.
Fast reactions, such as protein-protein interactions, can still be modelled as Boolean switches in these models, as they occur at completely different timescale.
A continuous model can be constructed  from the Boolean counterpart, when additional research is done through the measurement of kinetic data, followed by verification to measure the correspondence of the network with reality \cite{bolouri2002modeling}.
A proposed improvement upon the common ODE models is that of a sparse additive ODE model, able to capture nonlinear relationships \cite{wu2014sparse}.
Other types of models exist, notably stochastic models, hidden Markov models and multi-values Boolean models, but these fall outside the scope of this research \cite{bolouri2002modeling, wu2014sparse}.

The parameters in these models can be notoriously difficult to fit to emperical data, as this requires the search through multi-dimensional space for an optimum fit \cite{bolouri2002modeling, kuhn2009monte}.
This challenge has been tackled with, for instance, Monte Carlo methods with some success \cite{kuhn2009monte}.
More elaborate statistical methods have been developed as well, such as the modified elastic net-method, LASSO-methods and the Bayesian best subset regression \cite{greenfield2013robust, wu2014sparse}.

% Boolean networks
Peixoto so far
bivariate 21, 22
Justify continuous against RBN
They work under strong assumptions (for example inhibition dominates etc) \cite{}
We need an extension
Shlitt
Have to say, tononi1999measures suggest that robustness is due to degeneracy and redundancy, not due to synergy in neural networks

% ODE motifs
ferrell2011modeling
tyson2003sniffers
tyson2010functional
elowitz2000synthetic
goldbeter2002computational
hasty2001computational

% Full ODE
qian2008inference
wu2013high

\subsubsection{Information theory and GRNs}

An inquiry into GRNs using information theory has been made as well, primarily at the level of common network motifs \cite{zhang2012chaotic}.
Positive feedback loops have been found to function as switches and memory units, whereas negative feedback loops have been found to have a noise suppressing or oscilation-inducing function.
Studies have been done into the prevalence of chaotic behavior in GRNs \cite{zhang2012chaotic}.
These resulted in the conclusion that chaotic motifs of size $n\ge 3$ exist, but that they are uncommon in real networks, as they require competition between multiple feedback loops, at least one of which should be a negative feedback loop \cite{zhang2012chaotic}.
In real networks, this condition are scarcely met, although some GRNs do meet this condition, most notably the $n=4$ GRN that regulates the P53-system.

\subsubsection{Available GRN models}

There is a modest selection of GRN models available for further research.
These models are typically released in the SBML-format, an enriched xml-datatype.

One of the best captured and most researched GRNs is the endomesoderm GRN of the sea urchin \cite{bolouri2002modeling, kuhn2009monte}.
This network describes the activity of gene regulation in the early development of the sea urchin embryo, and is still in the proccess of being updated as new studies are done \cite{urchinmodel}.
An attemt has been made to build a full ODE model based on the Boolean abstraction of this GRN using Monte Carlo methods.
This was met with some success, as 65\% of the maximum possible correspondence with emperical data was achieved \cite{kuhn2009monte}.

% P53 is SMALL

% What datasets are out there?
%% ARTIFICIAL GRNs
synnetgen

ferrell2011modeling
aijo2009learning

\end{document}